src/pkg/backup/details/details.go

```diff
@@ -13,9 +13,76 @@ import (
 
 	"github.com/alcionai/corso/src/cli/print"
 	"github.com/alcionai/corso/src/internal/common"
+	"github.com/alcionai/corso/src/internal/version"
 	"github.com/alcionai/corso/src/pkg/path"
 )
 
+// UniqueLocationer provides access to location information but guarantees that
+// it can also generate a unique location (among items in the same service but
+// possibly across data types within the service) that can be used as a key in
+// maps and other structures. The unique location may be different than
+// DetailsLocation, the location used in backup details.
+type UniqueLocationer interface {
+	UniqueLocation() *path.Builder
+	DetailsLocation() *path.Builder
```

Comment:
These names are a bit of a mouthful.  Thoughts on this?
```suggestion
type LocationIDer interface {
	ID() *path.Builder
	InDetails() *path.Builder
```

Reply:
names are too difficult.  We should switch to emojis.